src/Uno.UI/UI/Xaml/Controls/Canvas/Canvas.Layout.cs

```diff
@@ -37,8 +37,8 @@ protected override Size ArrangeOverride(Size finalSize)
 
 				var childRect = new Rect
 				{
-					X = GetLeft(child as DependencyObject),
-					Y = GetTop(child as DependencyObject),
+					X = GetLeft((DependencyObject) child),
```

Comment:
This may break, a child may not be a `DependencyObject`.

Reply:
`foreach (var child in Children.Where(c => c is DependencyObject))`
:-)
No, that's not possible :-)